File: cathode/src/main/java/net/simonvt/cathode/ui/ShowsNavigationListener.java
/*
 * Copyright (C) 2013 Simon Vig Therkildsen
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.simonvt.cathode.ui;

import net.simonvt.cathode.common.ui.FragmentCallbacks;

public interface ShowsNavigationListener extends NavigationClickListener, FragmentCallbacks {

  void onDisplayShow(long showId, String title, String overview, LibraryType type);

  void onDisplaySeason(long showId, long seasonId, String showTitle, int seasonNumber,
      LibraryType type);

  void onDisplayEpisode(long episodeId, String showTitle);

  void onDisplayEpisodeHistory(long episodeId, String showTitle);

  void onDisplayRelatedShows(long showId, String title);

  void onSelectShowWatchedDate(long showId, String title);

  void onSelectSeasonWatchedDate(long seasonId, String title);

  void onSelectEpisodeWatchedDate(long episodeId, String title);

  void onSelectOlderEpisodeWatchedDate(long episodeId, String title);
}

File: cathode/src/main/java/net/simonvt/cathode/ui/ShowsNavigationListener.kt
/*
 * Copyright (C) 2013 Simon Vig Therkildsen
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.simonvt.cathode.ui

import net.simonvt.cathode.common.ui.FragmentCallbacks

interface ShowsNavigationListener : NavigationClickListener, FragmentCallbacks {

  fun onDisplayShow(showId: Long, title: String?, overview: String?, type: LibraryType)

  fun onDisplaySeason(
    showId: Long,
    seasonId: Long,
    showTitle: String?,
    seasonNumber: Int,
    type: LibraryType
  )

  fun onDisplayEpisode(episodeId: Long, showTitle: String?)

  fun onDisplayEpisodeHistory(episodeId: Long, showTitle: String)

  fun onDisplayRelatedShows(showId: Long, title: String?)

  fun onSelectShowWatchedDate(showId: Long, title: String?)

  fun onSelectSeasonWatchedDate(seasonId: Long, title: String?)

  fun onSelectEpisodeWatchedDate(episodeId: Long, title: String?)

  fun onSelectOlderEpisodeWatchedDate(episodeId: Long, title: String?)
}
